assembl/static2/js/app/components/cookies/acceptCookiesModal.jsx

```diff
@@ -0,0 +1,190 @@
+        acceptedLegalContentsList.includes(legalContent)
+      );
+    }
+    // The modal is only showed to a user who is connected but hasn't yet accepted legal contents and isn't currently reading them
+    if (!isOnLegalContentPage && !userHasAcceptedAllLegalContents && userId) {
+      this.setState({ showModal: true });
+    }
+  };
+
+  handleModalCheckbox = () => {
+    this.setState(prevState => ({
+      modalIsChecked: !prevState.modalIsChecked
+    }));
+  };
+
+  getLegalContentsToAccept = (): LegalContentsArray => {
+    const { hasTermsAndConditions, hasPrivacyPolicy, hasUserGuidelines } = this.props;
+    const legalContentsToAccept = {
+      ACCEPT_CGU: hasTermsAndConditions,
+      ACCEPT_PRIVACY_POLICY_ON_DISCUSSION: hasPrivacyPolicy,
+      ACCEPT_USER_GUIDELINE_ON_DISCUSSION: hasUserGuidelines
+    };
+    const filteredLegalContentsToAccept = LEGAL_CONTENTS_TO_ACCEPT.filter(contentType => legalContentsToAccept[contentType]);
+    return filteredLegalContentsToAccept;
+  };
+
+  acceptAllLegalContents = () => {
+    const legalContentsToAccept = this.getLegalContentsToAccept();
+    this.props.updateAcceptedCookies({
+      variables: { actions: legalContentsToAccept }
+    });
+  };
+
+  closeModal = () => {
+    this.setState({ showModal: false });
+  };
+
+  render() {
+    const { hasTermsAndConditions, hasPrivacyPolicy, hasUserGuidelines } = this.props;
+    const { showModal, modalIsChecked } = this.state;
+    const slug = getDiscussionSlug();
+    const legalContentsToAcceptByRouteName = {
+      terms: hasTermsAndConditions,
+      privacyPolicy: hasPrivacyPolicy,
+      userGuidelines: hasUserGuidelines
+    };
+    const legalContentsArray = Object.keys(legalContentsToAcceptByRouteName).map(
+      key => (legalContentsToAcceptByRouteName[key] ? key : null)
+    );
+    // This array gathers all the legal contents to accept by their route name
+    const cleanLegalContentsArray = legalContentsArray.filter(el => el !== null);
+
+    // TODO: deal with the backcross 'static' logic with this new Modal
+
+    return cleanLegalContentsArray ? (
+      <Modal show={showModal}>
+        <Modal.Header>
+          <Modal.Title>
+            <Translate value="legalContentsModal.title" />
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <FormGroup className="justify">
+            <Checkbox onChange={this.handleModalCheckbox} checked={modalIsChecked} type="checkbox" inline>
+              <Translate value="legalContentsModal.iAccept" /> {/* $FlowFixMe */}
+              <LegalContentsLinksList legalContentsList={cleanLegalContentsArray} />
+            </Checkbox>
+          </FormGroup>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button
+            key="cancel"
+            onClick={this.closeModal}
+            className="button-cancel button-dark"
+            href={`${getContextual('oldLogout', { slug: slug })}?next=${get('home', { slug: slug })}`}
```

Comment:
"old" logout?

Reply:
I don't know why it's called this way. There is only one logout route